src/index.js

```diff
@@ -1,52 +1,52 @@
 import React from "react"
 import PropTypes from "prop-types"
 
-function OutboundLink(props) {
+const NEWTAB = '_blank';
+const MIDDLE_CLICK = 1;
+
+function OutboundLink({ eventName, eventDetails, href, target, onClick, onMouseDown, children, ...rest }) {
+  const handleClick = e => {
+    const amplitudeEventName = eventName || window.amplitudeEventTypes.outboundLinkClick;
+    const amplitudeDetails = Object.assign({ href: href }, eventDetails)
+    const sameTarget = target !== NEWTAB;
+    const normalClick = !(e.ctrlKey || e.shiftKey || e.metaKey || e.button === MIDDLE_CLICK);
+
+    if (sameTarget && normalClick) {
+      e.preventDefault();
+      window.amplitude.getInstance().logEvent(amplitudeEventName, amplitudeDetails, () => {
+        window.location.href = href;
+      });
+    } else {
+      window.amplitude.getInstance().logEvent(amplitudeEventName, amplitudeDetails);
+    }
+
+    onClick && onClick(e);
+  };
+
+  // Adding mouseDown handler to track MIDDLE_CLICK, because MIDDLE_CLICK doesn't reach onClick event
```

Comment:
Most of this matches react-ga's OutboundLink except for this mouse down handler. Can you explain the inconsistency?

Reply:
When you click with the middle/wheel it doesn't reach the onClick event in react (at least in mac). That's why I added this custon onMouse logic, to be able to track middle/wheel clicks.

Reply:
So this is an improvement over react-ga's implementation? Just making sure. And you've tested that it only needs to be applied for middle click and not the other edge-case scenarios (meta/shift clicks)?

Reply:
yes, it is an improvement.

I've tested it with middle clicks, shift and ctrl keys in mac. (not windows)

I didn't test it with meta key, I don't know what key it is 😞

Reply:
I've tested the command click in mac, and it triggers the logEvent only once.

We can do another thing, we can remove the onMouseDown event. If someone wants to have this logic, he/she will be able to apply this custom logic in their own project.

What do you think?

Reply:
Yeah, I just am hesitant to add this unless I can be certain that it doesn't ever double-fire on any platform. It'd be great to reference some other popular open source project (like react-ga or another gatsby analytics plugin) that does this in a robust way.

Maybe at this point it'd be better to simplify and de-risk this PR so we can get the other things in? Can always add in this middle click behavior later once you're more confident in it.

Also, this is technically a breaking change, yes? Will need to do another major bump to 4.0.0, don't you think?